pkg/controller/dns/cleanup_runner.go

```diff
@@ -0,0 +1,29 @@
+package dns
+
+import (
+	"context"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type CleanupRunner struct {
+	client        client.Client
+	namesToDelete []string
+}
+
+var _ manager.Runnable = &CleanupRunner{}
+
+func newCleanupRunner(manager ctrl.Manager, namesToDelete []string) error {
+	runner := &CleanupRunner{
+		client:        manager.GetClient(),
+		namesToDelete: namesToDelete,
+	}
+	return manager.Add(runner)
+}
+
+func (r *CleanupRunner) Start(ctx context.Context) error {
+	// deletion would take place here using manifests.K8sNameKey
```

Comment:
I assume this is for external dns cleanup, are we holding off on that?

Reply:
yep! we will likely include that in a future PR as we're still fleshing out some of the details/implementation, but didn't want to just remove what had already been done since it doesn't interfere with current functionality